### ISSUE TITLE
Unable to open sideloaded zim file from website APK

### ISSUE THREAD

**Issue by Popolechien** [kiwix/kiwix-android#3993]
Downloaded the app from the website (Kiwix android 3.11.1 build 7231101) as well as a zim file from library.kiwix.org

When trying to open said zim file, get the following error message: `Selected Zim file could not be found`

Issue happened on GalaxyOS but I could reproduce on Android14
Here is the log:
[log.txt](https://github.com/user-attachments/files/16964761/log.txt)

**Comment by MohitMaliFtechiz** [comment]
@Popolechien 

* Can you please tell me which zim file you have downloaded and trying to open, and where is the sideloaded zim file located?

* Are you trying to open zim file via filePicker?

* Is this zim file showing in the LocalLibrary screen?

* In the `log.txt` there is only a zim file(WikEM) showing that should be opened, unless the file is not corrupted. 

```kotlin
/storage/emulated/0/Android/data/org.kiwix.kiwixmobile/files/Kiwix/wikem_en_all_maxi_2021-02.zim
```

**Comment by Popolechien** [comment]
* The zim file that was downloaded in both cases was https://download.kiwix.org/zim/wikivoyage/wikivoyage_en_europe_nopic_2024-06.zim, and the one tried by the other user who provided the log was https://download.kiwix.org/zim/wikivoyage/wikivoyage_en_all_maxi_2024-06.zim
* Tried to open the file via the apps filepicker (the big + in the library)
* The file is not showing in the app's library, but appears fine in the Download folder
* User had downloaded WikEM from the app just before trying to show me the sideloading issue. Adding my own log here:
[Diagnostic Report  .txt](https://github.com/user-attachments/files/16974430/Diagnostic.Report.txt)

**Comment by kelson42** [comment]
@MohitMaliFtechiz This issue is clearly the next priority.

**Comment by MohitMaliFtechiz** [comment]
@kelson42 I am on it.

**Comment by MohitMaliFtechiz** [comment]
@Popolechien Thanks for your feedback, and sorry for being late on this issue.

> The file is not showing in the app's library, but appears fine in the Download folder

Is the file not available after refreshing the local library(it should be available in the local library unless the file is not corrupted)? 

@Popolechien Since you are able to reproduce this error on your device, can you please test with this ZIM file and APK https://drive.google.com/file/d/1b0XbUQDxrWG3nV3_tPbx1xweBPoTKr-6/view?usp=sharing? and let me know if it is working or not?

**Comment by Popolechien** [comment]
Yep it worked... but then here I can not download the catalog (I get the infinite wheel just like in #3991, but that's a separate issue)

**Comment by MohitMaliFtechiz** [comment]
Thanks for confirming. I am sending you the latest website APK with https://github.com/kiwix/kiwix-android/issues/3991 fixes please try that APK and let us know so that we can finally close this issue.

**Comment by MohitMaliFtechiz** [comment]
@Popolechien Please try this APK it has the fixes of #3991 https://drive.google.com/file/d/1A97z5SFRp1bDYxEab9eZRlj932IoOVIF/view?usp=sharing.

**Comment by Popolechien** [comment]
Yup it works. Thanks.
